server/build/gzip.js

```diff
@@ -1,14 +1,26 @@
 import fs from 'fs'
 import path from 'path'
 import zlib from 'zlib'
+import rreaddir from 'recursive-readdir'
 
 export default async function gzipAssets (dir) {
   const nextDir = path.resolve(dir, '.next')
 
-  await Promise.all([
-    gzip(path.resolve(nextDir, 'commons.js')),
-    gzip(path.resolve(nextDir, 'main.js'))
-  ])
+  const coreAssets = [
+    path.join(nextDir, 'commons.js'),
+    path.join(nextDir, 'main.js')
+  ]
+  const pages = await getAllPages(nextDir)
+  const allAssets = [...coreAssets, ...pages]
+
+  while (true) {
+    // gzip only 10 assets in parallel at a time.
+    const currentChunk = allAssets.splice(0, 10)
```

Comment:
@rauchg it didn't sounds nice to gzip all at once in parallel to me.
Is this something I don't wanna worry?